### PR TITLE
ref(ui): Adjusting the width of things in the GlobalSelectionHeader

### PR DESCRIPTION
**Pull request by leedongwei**
## Main Header
Before (on wide screens):
image

After:
image

## Project Dropdown
Before:
image

After:
image

## Environment Dropdown
Before:
image

After:
image

## Date Dropdown
Before:
image
image

After:
image
image

Dropdown are set with to `min-width` so they can expand to fit more information as needed. 

`Do Not Merge` because I'll need to push to staging and check before merging into master.

### CODE REVIEW

**Comment by EvanPurkhiser** [comment]
> because I'll need to push to staging and check before merging into master

why not use the netlify previews 💯. Looks like you need added

**Comment by billyvg** [comment]
Can you post a screenshot with the dropdowns open too?

I think the max-widths were intentional so that they weren't stretched too far to the right on wide screens

**Comment by leedongwei** [comment]
@billyvg Updated PR comment with screenshots of the dropdowns.

**Review by ckj — APPROVED** [review]
Nothing out of place as far as I can tell. Thanks for taking the time to capture all the screenshots.

**Review by billyvg — APPROVED** [review]

**Comment by leedongwei** [comment]
> I think the max-widths were intentional so that they weren't stretched too far to the right on wide screens

@billyvg Sorry, missed out this comment earlier. Do you feel strongly about this? 

I'm fine with either `max-width: 450px` or `width: 33%` on columns. This PR was created because I was bothered by the dropdowns being wider than the columns. I can revert back if you prefer max-width.

**Comment by billyvg** [comment]
> > I think the max-widths were intentional so that they weren't stretched too far to the right on wide screens
> 
> @billyvg Sorry, missed out this comment earlier. Do you feel strongly about this?
> 
> I'm fine with either `max-width: 450px` or `width: 33%` on columns. This PR was created because I was bothered by the dropdowns being wider than the columns. I can revert back if you prefer max-width.

I think it might be better with `max-width` in the case where we only have 2 elements?

**Comment by billyvg** [comment]
@leedongwei damn now the mobile views are not right, let's just change it back to what you had previously w/ 33%

**Comment by billyvg** [comment]
@leedongwei o.o now we're missing some of the selectors... can we revert the last 3 commits and go with what you had before?

**Comment by leedongwei** [comment]
- Rather than declaring max-width/width of the `HeaderItemPosition`, I've added empty `HeaderItemPosition` when environment/datetime is not required on the header and let flex-box work its magic.
- Plucked some low-hanging fruits for Typescript

File: src/sentry/static/sentry/app/components/organizations/headerItem.tsx [hunk 1]
```diff
@@ -158,4 +173,6 @@ const StyledLock = styled(InlineSvg)`
   stroke-width: 1.5;
 `;
 
-export default React.forwardRef((props, ref) => <HeaderItem {...props} innerRef={ref} />);
+export default React.forwardRef((props: Props, ref: React.Ref<any>) => (
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/organizations/headerItem.tsx]: ref gets passed to `StyledHeaderItem` - so should it be `React.Ref<HTMLDivElement>`?

**Review by billyvg — APPROVED** [review]
image

Should the menu line up with separator?

**Comment by leedongwei** [comment]
@billyvg I've looked into that when tweaking things but it's not worth chasing that 1px separator because I'll need to write separate CSS rules for the first/last/middle `HeaderItemPosition`.

**Comment by leedongwei** [comment]
Requesting for new review as I had to rebase against `master` and fix linter complaint.

**Review by billyvg — APPROVED** [review]
Looks good, thanks

In the future, can you wait to approve on Percy until you're ready to merge PR? This helps reviewers (maybe it's just me) realize that there are Percy diffs to review.

**Pull request merged**